core/src/test/scala/integration/kafka/api/EndToEndAuthorizationTest.scala

```diff
@@ -352,20 +359,28 @@ abstract class EndToEndAuthorizationTest extends IntegrationTestHarness with Sas
     // Verify successful produce/consume/describe on another topic using the same producer, consumer and adminClient
     val topic2 = "topic2"
     val tp2 = new TopicPartition(topic2, 0)
+
     setReadAndWriteAcls(tp2)
-    sendRecords(producer, numRecords, tp2)
+    // in idempotence producer, we need to create another producer because the previous one is in FATEL_ERROR state (due to authorization error)
```

Comment:
`FATEL_ERROR` -> `FATAL_ERROR`
Do we really expect the idempotent producer to not be usable anymore?

Reply:
I think that's how it is designed. I added more comment for it to make it clear:
`//     // in idempotence producer, we need to create another producer because the previous one is in FATEL_ERROR state (due to authorization error)`
`// If the transaction state in FATEL_ERROR, it'll never transit to other state. check TransactionManager#isTransitionValid for detail`

Thank you.

Reply:
Let's still fix the typo, it should be `FATAL_ERROR`

Reply:
ah, you're right, I forgot it.